astropy/cosmology/parameter/_core.py

```diff
@@ -7,14 +7,24 @@
 import copy
 from dataclasses import KW_ONLY, dataclass, field, fields, is_dataclass, replace
 from enum import Enum, auto
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Generic, TypeVar, overload
 
 import astropy.units as u
 
-from ._converter import _REGISTRY_FVALIDATORS, FValidateCallable, _register_validator
+from ._converter import _REGISTRY_FVALIDATORS, _register_validator
 
 if TYPE_CHECKING:
-    from collections.abc import Sequence
+    from collections.abc import Callable, Sequence
+    from typing import TypeAlias
+
+    from typing_extensions import Self
+
+    from astropy.cosmology import Cosmology
+
+    _FValidateCallable: TypeAlias = Callable[[Cosmology, "Parameter", Any], "_VT"]
```

Comment:
```suggestion
    _FValidateCallable: TypeAlias = Callable[[Cosmology, Parameter, Any], _VT]  # noqa: F821
```

Reply:
Let's see...

Reply:
Now you've undone this change.

Reply:
This alias is used in more than one file, so perhaps we should define it in a separate `typing` module. That module should be able to import `Parameter` without triggering Ruff F821 or mypy `used-before-def`.

Reply:
SGTM. I'll make a `_typing.py` module. Private for now. I would be interested to make it public once it's stabilized a bit.

Reply:
Done!